src/services/json-api-datastore.service.ts

```diff
@@ -208,19 +208,34 @@ export class JsonApiDatastore {
 
     private extractRecordData<T extends JsonApiModel>(res: Response, modelType: ModelType<T>, model?: T): T {
         let body: any = res.json();
+
         if (!body) {
             throw new Error('no body in response');
         }
+
         if (model) {
             model.id = body.data.id;
             Object.assign(model, body.data.attributes);
         }
+
         model = model || this.deserializeModel(modelType, body.data);
         this.addToStore(model);
         if (body.included) {
             model.syncRelationships(body.data, body.included, 0);
             this.addToStore(model);
+
+            body.included.forEach((includedModelData: any) => {
+                const models = Reflect.getMetadata('JsonApiDatastoreConfig', this.constructor).models;
+
+                const options = new ResponseOptions({
+                    body: JSON.stringify({data: includedModelData})
+                });
+                const includedModelResponse: Response = new Response(options);
+
+                this.extractRecordData(includedModelResponse, models[includedModelData.type]);
```

Comment:
IMO an overload for `extractRecordData´ that accepts the plain data object would be better.